scripts/Test-RepoPackage.ps1

```diff
@@ -0,0 +1,633 @@
+  in the repository before running the tests.
+
+.PARAMETER TakeScreenshots
+  If this variable have been provided, the script will take a screenshot
+  of the user desktop after install/uninstall as well as when the message
+  'Failures' have been outputted by the choco executable.
+
+.PARAMETER chocoCommandTimeout
+  The timeout that should be passed to choco when installing/uninstalling
+  packages (defaults to 600 seconds, or 10 minutes)
+
+.PARAMETER timeoutBeforeScreenshot
+  The amount of seconds to sleep after installing/uninstalling a package
+  and before taking the screenshot.
+
+.PARAMETER screenShotDir
+  The directory to where the scrip should save all taken screenshots.
+  (The directory does not need to exist before running the script)
+
+.EXAMPLE
+  .\Test-RepoPackage.ps1
+#>
+param(
+  [string]$packageName = $null,
+  [ValidateSet('all','update','install','uninstall','none')]
+  [string]$type = 'all',
+  [switch]$CleanFiles,
+  [switch]$TakeScreenshots,
+  [int]$chocoCommandTimeout = 600,
+  [int]$timeoutBeforeScreenshot = 1,
+  [string]$screenShotDir = "$env:TEMP\screenshots"
+)
+
+function WriteOutput() {
+<#
+.SYNOPSIS
+  Simple helper function to simplify
+  the different output methods we use.
+#>
+  param(
+    [Parameter(ValueFromPipeline = $true)]
+    [string]$text,
+    [ValidateSet('Error','Warning','ChocoError','ChocoWarning','ChocoInfo','Normal')]
+    [string]$type = 'Normal'
+  )
+   switch -Exact ($type) {
+    'Error' { Write-Error $text }
+    'Warning' { Write-Warning $text }
+    'ChocoError' { Write-Host $text -ForegroundColor Black -BackgroundColor Red }
+    'ChocoWarning' { Write-Host $text -ForegroundColor Black -BackgroundColor Yellow }
+    'ChocoInfo' { Write-Host $text -ForegroundColor White -BackgroundColor Black }
+    Default { Write-Host $text }
+  }
+}
+
+function WriteChocoOutput() {
+<#
+.SYNOPSIS
+  A simple helper function for processing the output of choco.
+  Only meant to handle the colorization of text.
+#>
+  param(
+    [Parameter(ValueFromPipeline = $true)]
+    [string]$text
+  )
+  begin {}
+  process {
+    switch -Regex ($text) {
+      '^\s*ERROR|\s*Failures|^Uninstall may not be silent' { WriteOutput $text -type ChocoError }
+      '^\s*WARNING' { WriteOutput $text -type ChocoWarning }
+      Default { WriteOutput $text -type ChocoInfo }
+    }
+  }
+  end {}
+}
+
+function CleanFiles() {
+<#
+.SYNOPSIS
+  The function responsible for cleaning out temporary
+  files before we run the tests.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    [string]$screenShotDir
+  )
+  $pathsToClean = @(
+    "$env:ChocolateyInstall\logs\*"
+    "$env:TEMP\chocolatey\au\*"
+    "$PSScriptRoot\..\Update-Force-Test*.md"
+  )
+  if ($TakeScreenshots) { $pathsToClean += @("$screenShotDir\*") }
+  $pathsToClean += Get-ChildItem -Recurse "$PSScriptRoot\.." -Filter "*.nupkg" | select -Expand FullName
+
+  $pathsToClean | % {
+    if (Test-Path $_) { rm $_ -Recurse }
```

Comment:
I did try that first, however it deleted a few files that it wasn't supposed to delete.
I still can't figure out why though, but worked as expected when I made it like this.

Also, I believe using -ea 0 would be incorrect, as we would probably want to error out if a file isn't possible to delete, like if it's locked or if it requires some permissions that the user don't have.